runtime/flang/fpcvt.c

```diff
@@ -1172,8 +1289,13 @@ __fortio_fcvt(__BIGREAL_T v, int prec, int sf, int *decpt, int *sign, int round)
         128, 132, 136, 140, 144, 148, 152, 155, 159, 163, 167, 170, 174,
         178, 181, 185, 188, 192, 195, 198, 202, 205, 208, 212, 215, 218,
         221, 224, 228, 231, 234, 237, 240, 243, 246, 249, 252, 255};
-      nexp = ((ieee_v.i[1] & 0x7ff00000) >> 12) - 261881; // 261880 works too
-      idx = ((ieee_v.i[1] & 0xfc000) >> 14);
+      if (!is_quad) {
+        nexp = ((ieee_v.i[1] & MASK_EXPONENT_DBLE) >> SHIFT_EXPONENT_DBLE) - BIAS_DBLE; // 261880 works too
```

Comment:
with this change, we no longer see to which part of the expression this `261880` refers to...

Reply:
Ok, already back to the original.